add: Обновление кровотечений (часть 2)

vovakons opened the pull request:
## Что этот ПР делает
1. Добавлены артериальные кровотечения (*условия получия такие же как и у внутряков, но шанс чуть ниже*)
2. Добавлен самодельный жгут, жгут и турникет для остановки кровотечений на конечностях (*если зажать жгутом конечность слишком долго - некроз конечности*).
3. Добавление новых предметов в вендоры.
4. Показ кровотечения в анализаторах здоровья (*ручной и машинерия*)
5. Бинты и наборы для сшивания ран теперь работают сразу рука+ладошка и нога+пятка (*то есть наложение на ногу остановит кровотек и на пятке*)
6. Замедление скорости потери кровви в зависимости от температуры тела и объема крови в теле.
7. Жгуты снимаются через self check или через шифт клик.
8. Внешнее кровотечение немного ослаблено.
9. Артериальное кровотечение имеет эффект струй крови которое даст визаульное обозначение мобов с артериальным кровотечением.
10. Артериальное кровотечение является смертельной (*моб может вытечь в течении минуты если не наложить жгут или не провести операцию*)
11. Артериальное кровотечение лечится также как и внутряк операцией "восстановление сосудов".
12. Артериальных кровотечений не может быть на пятках и ладошках.
13. Эффект запутанности ног при малом количестве крови в теле (*не часто*)


## Почему это хорошо для игры

Дальнешее улучшение механики кровотеков.

## Демонстрация изменений
<img width="331" height="257" alt="image" src="https://github.com/user-attachments/assets/18cae9c1-21ad-46a7-8de3-e6b26edd691c" />
<img width="669" height="220" alt="image" src="https://github.com/user-attachments/assets/102cd856-d56a-4399-8680-a3560c5f42a6" />
<img width="478" height="267" alt="image" src="https://github.com/user-attachments/assets/a0faf150-8466-46a2-a775-20ccf7b34b86" />
<img width="486" height="155" alt="image" src="https://github.com/user-attachments/assets/b41afcaf-bea3-45da-888c-40c6c681e074" />



## Тестирование

Локалка.

vovakons committed:
suppress bleeding work in arms+hands and legs+foors

Теперь бинты и наборы для сшивания ран останавливают кровотечение на
конечности сразу парой.
То есть рука + ладошка сразу, нога и пятка.

vovakons committed:
no blood trait organs no bleeding

Убрал кровотечение у мобов у которых есть трейт NO_BLOOD.

vovakons committed:
show bleeding in health analyzer

Показ места кровотечения в анализаторе здоровья.

vovakons committed:
bleeding in body scanner

Показ места и силы кровотечения в BodyScanner.

vovakons committed:
arterial bleeding begin

Добавлено артериальное кровотечение через обычное кровотечение.

vovakons committed:
arterial bleeding iteration 1

vovakons committed:
tourniquets

Добавлен самодельный жгут, жгут и турникет.
Они позволяют полностью закрыть любое кровотечение на конечностях.

vovakons committed:
confused if low blood

vovakons committed:
blood lose speed by blood volume

Добавлен модификатор скорости потери крови в зависимости от количества
крови в теле. Чем меньше кровь в теле - тем меньше скорость потери
крови.

vovakons committed:
blood lose speed by volume convert to percentages

vovakons committed:
blood loss by body temperature

Скорость потери крови зависит от температуры тела. При низкой
температуре тела теряется меньше крови.

vovakons committed:
oxy damage if tourniquet aapply on head

Добавляет окси урон если применить жгут на голову (шею).

vovakons committed:
necrotize limb with tourniquet long delay

После длительного ношения жгута на конечности происходит некроз
конечности.
Предупреждение за минуту до некроза.
Некроз через 5 минут.

vovakons committed:
tourniquet remove from shift click

vovakons committed:
add tourniquets in vendings and add craft

Добавлена возможность создавать самодельные жгуты из проводов.
Добавлены жгут и турникет в различные наномеды.

pilygun commented:
Мне кажется это уже перебор, звучит душновато

PlayerUnknown14 commented:
> Мне кажется это уже перебор, звучит душновато

Нахуй казуальность, движемся к тарков-стейшен на трейтьей космической.

tsoivecen199015 commented:
> Мне кажется это уже перебор, звучит душновато

Всегда можно ввести нанитов что залатают этот тип кровотёка за 5 секунд

UnderwaterNeike commented:
Так мы готовим секретный билд, аналог «Бургер Стейшн», к релизу Таркова, чтобы быть на хайпе.🐈💨

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
first aid fixes

Возможность хранить жгуты в аптечке.
Количество слотов аптечки увеличено до 9 штук под новые медицинские
штуки.

vovakons committed:
run tgui prettier

vovakons committed:
ballon_alert fix

vovakons committed:
defines check

vovakons committed:
blood system defines fix

Ksaikok46 commented on code/game/machinery/adv_med.dm:
```diff
@@ -525,10 +528,22 @@
+			if(e.bleeding_amount > 0)
+				bleeding = "<br>"
+				if(e.bleeding_amount <= e.bleedsuppress)
+					bleeding += "перевязанное "
```
Может быть лучше "остановленное кровотечение", а не "перевязанное кровотечение"?

Ksaikok46 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -749,20 +753,33 @@ BODY SCANNERS
+		if(e.bleeding_amount > 0)
+			var/bleeding = "[limb] -"
+			if(e.bleeding_amount <= e.bleedsuppress)
+				bleeding += " перевязаное "
```
Здесь аналогично, "кровотечение в груди - остановлено"(либо остановленное), а не "кровотечение в груди - перевязаное"? И кажется пробел в конце слова лишний

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+	if(human_target == user)
+		user.visible_message(
+			span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять [declension_ru(NOMINATIVE)] на себе."),
+			span_notice("Вы начинаете применять [declent_ru(NOMINATIVE)] на себе..."),
```
Не лучше ли здесь и далее "применять [declent_ru(NOMINATIVE)] на себе" -> "применять на себе [declent_ru(ACCUSATIVE)]"?

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+	user.drop_item_ground(src)
+	src.forceMove(affecting)
+
+	balloon_alert(user, "жгут наложен.")
```
Здесь и далее у балун алертов точки не нужны в конце

Ksaikok46 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -376,16 +376,24 @@
+			continue
+		if(bodypart.tourniquet.applyed_bodypart != bodypart)
+			continue
+		msg += span_warning("<a href='byond://?src=[UID()];tourniquet_object=[bodypart.tourniquet.UID()];limb=[bodypart.UID()]' class='warning'>У [genderize_ru(gender, "него", "неё", "него", "них")] [bodypart.declent_ru(NOMINATIVE)] перевязан[genderize_ru(gender, "", "а", "о", "ы")] [bicon(bodypart.tourniquet)] [bodypart.tourniquet.declent_ru(INSTRUMENTAL)]!</a>\n")
```
Не лучше ли "У него правая нога перевязана жгутом!" -> "Его правая нога перевязана жгутом!"?

Ksaikok46 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -376,16 +376,24 @@
 			msg += span_warning("У [genderize_ru(gender, "него", "неё", "него", "них")] [bodypart.declent_ru(NOMINATIVE)] перевязан[genderize_ru(gender, "", "а", "о", "ы")] чем-то окровавленным.\n")
-		else if (high_bleeding)
+		else if(bodypart.has_arterial_bleeding())
+			msg += span_warning(span_bold("У [genderize_ru(gender, "него", "неё", "него", "них")] кровь течет фонтаном из [bodypart.declent_ru(ACCUSATIVE)]!\n"))
```
Не лучше ли
"У него кровь течет фонтаном из declent_ru(ACCUSATIVE)!" -> "Из его declent_ru(GENITIVE) хлещет кровь!"

Ksaikok46 commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -980,6 +1011,48 @@ Note that amputating the affected organ does in fact remove the infection from t
+
+	if(owner && !silent)
+		owner.custom_pain("Вы чувствуете, как из ваш[genderize_ru(gender, "ем", "ей", "ем", "ем")] [declent_ru(PREPOSITIONAL)] кровь хлещет фонтаном!")
+		owner.visible_message(span_warning("У [owner] фонтаном хлещет кровь из [declent_ru(PREPOSITIONAL)]!"))
```
"из ваш(ем, ей, ем, ем) declent_ru(PREPOSITIONAL) кровь хлещет фонтаном!" -> "из ваш(его, ей, его, их) declent_ru(GENITIVE) хлещет кровь!
И ниже также поправить

Ksaikok46 commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -326,11 +326,12 @@
 				status += " перевязан[genderize_ru(bodypart.gender, "", "а", "о", "ы")] чем-то окровавленным"
-			else if(high_bleeding)
+			else if(bodypart.has_arterial_bleeding())
+				status += " кровь течет фонтаном"
```
"хлещет кровь"?

Ksaikok46 reviewed (commented):
Предложения по правкам буков

vovakons committed:
review fixes

vovakons committed:
rebuild tgui

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -56,7 +56,7 @@
 			user.visible_message(
 				span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять [declension_ru(NOMINATIVE)] на себе."),
-				span_notice("Вы начинаете применять [declent_ru(NOMINATIVE)] на себе..."),
+				span_notice("Вы начинаете применять на себе [declent_ru(NOMINATIVE)]..."),
```
```suggestion
				span_notice("Вы начинаете применять на себе [declent_ru(ACCUSATIVE)]..."),
```

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+	var/selected_zone = user.zone_selected
+	if(human_target == user)
+		user.visible_message(
+			span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять [declension_ru(NOMINATIVE)] на себе."),
```
```suggestion
			span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять на себе [declension_ru(ACCUSATIVE)]."),
```

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+	if(human_target == user)
+		user.visible_message(
+			span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять [declension_ru(NOMINATIVE)] на себе."),
+			span_notice("Вы начинаете применять на себе [declent_ru(NOMINATIVE)]..."),
```
```suggestion
			span_notice("Вы начинаете применять на себе [declent_ru(ACCUSATIVE)]..."),
```

Ksaikok46 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -909,7 +927,19 @@ BODY SCANNERS
+			else if(e.has_heavy_bleeding())
+				bleed_stat += "обильное "
+			if(e.bleeding_amount <= e.bleedsuppress)
+				bleed_stat += "перевязанное "
```
```suggestion
				bleed_stat += "остановленное "
```

Ksaikok46 commented on code/game/machinery/adv_med.dm:
```diff
@@ -337,6 +337,9 @@
 			organData["bruised"] = E.min_bruised_damage
 			organData["broken"] = E.min_broken_damage
+			organData["bleed"] = round(E.bleeding_amount, 0.01)
+			organData["bleed_supp"] = "[(E.bleeding_amount <= E.bleedsuppress) ? "перевязано" : ""]"
```
```suggestion
			organData["bleed_supp"] = "[(E.bleeding_amount <= E.bleedsuppress) ? "остановлено" : ""]"
```

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -56,7 +56,7 @@
 		if(human_target == user && !unique_handling)
 			user.visible_message(
 				span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять [declension_ru(NOMINATIVE)] на себе."),
```
```suggestion
				span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять на себе [declension_ru(ACCUSATIVE)]."),
```

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+			to_chat(human_target, span_danger("[capitalize(declent_ru(NOMINATIVE))] нельзя применить на протезе!"))
+			return .
+		user.visible_message(
+			span_green("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(NOMINATIVE)] на [human_target.declent_ru(NOMINATIVE)]."),
```
```suggestion
			span_green("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(ACCUSATIVE)] на [human_target.declent_ru(ACCUSATIVE)]."),
```

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+			return .
+		user.visible_message(
+			span_green("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(NOMINATIVE)] на [human_target.declent_ru(NOMINATIVE)]."),
+			span_green("Вы применяете [declent_ru(NOMINATIVE)] на [human_target.declent_ru(NOMINATIVE)]."),
```
```suggestion
			span_green("Вы применяете [declent_ru(ACCUSATIVE)] на [human_target.declent_ru(ACCUSATIVE)]."),
```

Ksaikok46 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -376,16 +376,24 @@
-		var/high_bleeding = bodypart.bleeding_amount > HIGH_BLEEDING_VALUE
 		var/suppressed = bodypart.bleeding_amount <= bodypart.bleedsuppress
 		if(suppressed)
 			msg += span_warning("У [genderize_ru(gender, "него", "неё", "него", "них")] [bodypart.declent_ru(NOMINATIVE)] перевязан[genderize_ru(gender, "", "а", "о", "ы")] чем-то окровавленным.\n")
```
```suggestion
			msg += span_warning("[capitalize(genderize_ru(gender, "его", "её", "его", "их"))] [bodypart.declent_ru(NOMINATIVE)] перевязан[genderize_ru(bodypart.gender, "", "а", "о", "ы")] чем-то окровавленным.\n")
```

Ksaikok46 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -376,16 +376,24 @@
 			msg += span_warning("У [genderize_ru(gender, "него", "неё", "него", "них")] [bodypart.declent_ru(NOMINATIVE)] перевязан[genderize_ru(gender, "", "а", "о", "ы")] чем-то окровавленным.\n")
-		else if (high_bleeding)
+		else if(bodypart.has_arterial_bleeding())
+			msg += span_warning(span_bold("Из [genderize_ru(gender, "его", "её", "его", "их")] [bodypart.declent_ru(GENITIVE)]  хлещет кровь!\n"))
```
```suggestion
			msg += span_warning(span_bold("Из [genderize_ru(gender, "его", "её", "его", "их")] [bodypart.declent_ru(GENITIVE)] хлещет кровь!\n"))
```

Ksaikok46 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -376,16 +376,24 @@
-		else if (high_bleeding)
+		else if(bodypart.has_arterial_bleeding())
+			msg += span_warning(span_bold("Из [genderize_ru(gender, "его", "её", "его", "их")] [bodypart.declent_ru(GENITIVE)]  хлещет кровь!\n"))
+		if (bodypart.has_heavy_bleeding())
```
```suggestion
		if(bodypart.has_heavy_bleeding())
```

Ksaikok46 commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -980,6 +1011,48 @@ Note that amputating the affected organ does in fact remove the infection from t
+	INVOKE_ASYNC(owner, TYPE_PROC_REF(/mob, emote), "scream")
+
+	if(owner && !silent)
+		owner.custom_pain("Из ваш[genderize_ru(gender, "ем", "ей", "ем", "ем")] [declent_ru(GENITIVE)] хлещет кров!")
```
```suggestion
		owner.custom_pain("Из ваш[genderize_ru(gender, "его", "ей", "его", "их")] [declent_ru(GENITIVE)] хлещет кровь!")
```

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+			return .
+	else
+		user.visible_message(
+			span_notice("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(NOMINATIVE)] на [human_target]."),
```
```suggestion
			span_notice("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(ACCUSATIVE)] на [human_target]."),
```

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+	else
+		user.visible_message(
+			span_notice("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(NOMINATIVE)] на [human_target]."),
+			span_notice("Вы начинаете применять [declent_ru(NOMINATIVE)] на [human_target]..."),
```
```suggestion
			span_notice("Вы начинаете применять [declent_ru(ACCUSATIVE)] на [human_target]..."),
```

Ksaikok46 reviewed (commented):
В остальном по буквам вроде порядок

littleboobs commented on code/game/gamemodes/miniantags/guardian/types/healer.dm:
```diff
@@ -209,7 +209,9 @@
-					to_chat(user, "Внутреннее кровотечение остановлено.")
+					limb.stop_arterial_bleeding()
+					limb.stop_bleeding()
+					to_chat(user, "Кровотечение остановлено.")
```
это балун?

vovakons replied:
неа, это голопара пишет в чат про лечение

littleboobs commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -749,20 +753,34 @@ BODY SCANNERS
 		if(!e)
 			continue
-		var/limb = e.name
+		var/limb = e.declent_ru(PREPOSITIONAL)
```
имя переменной, конечно.....

littleboobs commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -749,20 +753,34 @@ BODY SCANNERS
+				bleeding += "Кровотечение"
+			bleeding += "  в [limb]"
+			if(e.bleeding_amount <= e.bleedsuppress)
+				bleeding += " - остановлено"
```
```suggestion
				bleeding += " – остановлено"
```

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
 	icon_state = "synthkit_[amount]"
+
+
+// MARK: Tourniquet
+
+/obj/item/tourniquet
```
```suggestion

// MARK: Tourniquet
/obj/item/tourniquet
```

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+	var/necrotize_warning_duration = 4 MINUTES
+	var/necrotize_warning_timer_id = null
+	var/necrotize_duration = 5 MINUTES
+	var/necrotize_timer_id = null
```
комментарии нужны

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+		user.visible_message(
+			span_notice("[human_target] начина[pluralize_ru(human_target.gender,"ет","ют")] применять [declension_ru(NOMINATIVE)] на себе."),
+			span_notice("Вы начинаете применять на себе [declent_ru(NOMINATIVE)]..."),
+		)
```
Делай подобное в таких конструкциях
<img width="750" height="123" alt="image" src="https://github.com/user-attachments/assets/53d73f14-0e76-4734-9d4d-78aa9a674f0a" />

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+		user.visible_message(
+			span_notice("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(NOMINATIVE)] на [human_target]."),
+			span_notice("Вы начинаете применять [declent_ru(NOMINATIVE)] на [human_target]..."),
+		)
```
пример выше

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -701,3 +707,201 @@
+		user.visible_message(
+			span_green("[user] применя[pluralize_ru(user.gender,"ет","ют")] [declent_ru(NOMINATIVE)] на [human_target.declent_ru(NOMINATIVE)]."),
+			span_green("Вы применяете [declent_ru(NOMINATIVE)] на [human_target.declent_ru(NOMINATIVE)]."),
+		)
```
пример выше

littleboobs commented on code/modules/power/cable_coil.dm:
```diff
@@ -140,6 +142,12 @@
 				cablecuff.icon_state = "cuff_[text_color]"
+		if("самодельный жгут (20)")
+			if(get_amount() < 20)
+				to_chat(user, span_warning("Вам не хватает [declent_ru(GENITIVE)] чтобы сделать самодельный жгут!"))
```
балун? "недостаточно хуйнюшки"

littleboobs commented on code/modules/surgery/organs/blood.dm:
```diff
@@ -1,4 +1,46 @@
+// Blood system
+// MARK: Definitions
+
+#define EXOTIC_BLEED_MULTIPLIER 8 //Multiplies the actually bled amount by this number for the purposes of turf reaction calculations.
```
поправь комм

littleboobs reviewed (changes requested):
Оставил комментарии некоторые. Спасибо господину КсайКоку, что сделал ревью. Его комментарии тоже надо поправить. В остальном нормально.

vovakons committed:
localization update

vovakons committed:
review fixes

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
arterial bleeding size balance

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
linter fixes 1

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
rebuild tgui

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
rebuild tgui

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Update tgui.bundle.js

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
decrease suture kit damage

Уменьшен урон при зашивании набором с 10 до 5 брута.

vovakons committed:
tourniquet necrotize duration decrease

Уменьшено время некроза конечности при наложении жгута с 5 минут до 3
минут.
Уменьшено время предупреждения о некрозе с 4 минут до 2 минут.

vovakons committed:
add neck cut feature for all knives

Добавлен функционал перерезания горла при красном захвате и ударе ножом
в харме в голову.
При перерезании горла цель теряет 25% своей крови и получает
артериальное кровотечение в голове.
С исскуством ножевого боя перерезание горла происходит за 1 секунду, без
исскуства 5 секунд.

vovakons committed:
supress bleeding by hands

Добавлено зажимание кровотечений руками.

vovakons committed:
bruise pack supress bledding to zero

Теперь бинты полностью закрывают кровотечение, удалена логика с
процентным остановлением кровотека.

vovakons committed:
rebuild tgui

vovakons committed:
add undef for hand grab suppress bloodloss

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Update tgui.bundle.js

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
Merge branch 'bleeding-part-2' of github.com:vovakons/Paradise into bleeding-part-2

vovakons committed:
rebuild tgui

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
update tgui

dageavtobusnick committed:
Merge branch 'master220' into bleeding-part-2

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
фикс

dageavtobusnick committed:
фикс

vovakons commented:
ахуй, дожили 😃

vovakons committed:
Merge branch 'master220' into bleeding-part-2

Ksaikok46 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,210 @@
+			to_chat(human_target, span_danger("[capitalize(declent_ru(NOMINATIVE))] нельзя применить на протезе!"))
+			return .
+	else
+		balloon_alert(user, "применение [declent_ru(GENITIVE)]...")
```
Что думаешь насчёт замены этого балуна на balloon_alert_to_viewers, чтобы всем было видно что кто-то накладывает жгут?

vovakons committed:
tourniquet apply fixes

Увеличена длительность наложения жгута.
Показ ballon для всех при наложении жгута.
Красный текст в чате про наложенный жгут.
Уменьшение окси урона от жгута в голову до 4 в тик.

vovakons committed:
take arterial bleeding chance rework

Изменение логики накладывания артериального кровотечения, шанс зависит
от итогового урона части тела и урона оружия.
Острые предметы имеют модификатор 1.25 к шансу нанесения артериального
кровотека.
Слегка увеличена кровопотеря.

vovakons committed:
reduce hand bleed suppress size

Максимальное значение зажатия кровотечения рукой - 15 кровотечения.

dageavtobusnick committed:
Merge branch 'master220' into bleeding-part-2

dageavtobusnick committed:
Merge branch 'master220' into bleeding-part-2

dageavtobusnick committed:
Merge branch 'master220' into bleeding-part-2

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Update tgui.bundle.js

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
rebuild tgui

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
temp remove tourniquets from game

Удаление жгутов в щитспавн, временно, пока не будет третья обнова на
кровотеки.

vovakons committed:
arterial bleeding chance rework

Изменение логики получения артериального кровотечения - условия такие же
что и внутряки, но шанс в два раза ниже.

vovakons committed:
brute damage for neck cut

Добавлено 30 брут урона при перерезании горла ножом.
Теперь можно перерезать горло несколько раз.

shahribek commented:
самодельные жгуты это хорошо. теперь ждём самодельные турникеты

vovakons committed:
self suppress with alt click on self

Зажатие раны руками перенесено на альт клик вместо граба по себе.

vovakons committed:
kitchen knife with neck cut no more attack target

vovakons committed:
cut all tourniquets by wirecutters

Клик кусачками по любому персонажу с жгутами срезает все жгуты за один
клик.

vovakons committed:
Merge branch 'master220' into bleeding-part-2

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -54,8 +54,8 @@
-				span_notice("Вы начинаете применять [declent_ru(NOMINATIVE)] на себе..."),
+				span_notice("[human_target] начина[PLUR_ET_YUT(human_target)] применять на себе [declent_ru(ACCUSATIVE)]."),
+				span_notice("Вы начинаете применять на себе [declent_ru(ACCUSATIVE)]..."),
 			)
```
замени на балун ту вьюерс

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+		user.visible_message(
+			span_notice("[human_target] начина[PLUR_ET_UT(human_target)] применять на себе [declent_ru(ACCUSATIVE)]."),
+			ignored_mobs = user
+		)
```
замени на балун ту вьюерс

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+			balloon_alert(user, "уже наложен жгут!")
+			return .
+		if(affecting_rechecked.is_robotic())
+			to_chat(human_target, span_danger("[capitalize(declent_ru(NOMINATIVE))] нельзя применить на протезе!"))
```
балун

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+		user.visible_message(
+			span_danger("[user] применя[PLUR_ET_UT(user)] [declent_ru(ACCUSATIVE)] на [human_target]."),
+			ignored_mobs = user
+		)
```
замени на балун ту вьюерс

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+			balloon_alert(user, "уже наложен жгут!")
+			return .
+		if(affecting_rechecked.is_robotic())
+			to_chat(human_target, span_danger("[capitalize(declent_ru(NOMINATIVE))] нельзя применить на протезе!"))
```
балун

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+		user.visible_message(
+			span_danger("[user] применя[PLUR_ET_UT(user)] [declent_ru(ACCUSATIVE)] на [human_target.declent_ru(ACCUSATIVE)]."),
+			span_green("Вы применяете [declent_ru(ACCUSATIVE)] на [human_target.declent_ru(ACCUSATIVE)]."),
+		)
```
замени на балун ту вьюерс

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+/obj/item/tourniquet/proc/remove_from_bodypart(mob/living/user)
+	if(!applyed_bodypart)
+		return FALSE
+	balloon_alert(user, "снятие жгута...")
```
замени на балун ту вьюерс с сообщением для всех о снятии жгута

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+	user.drop_item_ground(src)
+	src.forceMove(affecting)
+
+	user.balloon_alert_to_viewers("жгут наложен")
```
второе сообщение потерял

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+		applyed_addition_bodypart = null
+	stop_apply_timers()
+	user.put_in_any_hand_if_possible(src)
+	balloon_alert(user, "жгут снят")
```
выше у тебя используется балун ту вьюерс, разберись вообще, че ты и кому показывать хочешь

littleboobs commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -672,3 +678,232 @@
+
+/mob/living/carbon/human/proc/cut_all_tourniquets(mob/living/user)
+	balloon_alert(src, "жгуты срезаны!")
+	balloon_alert(user, "жгуты срезаны!")
```
???

littleboobs commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -187,15 +187,63 @@
+	if(selected_zone != BODY_ZONE_HEAD)
+		return FALSE
+	var/obj/item/organ/external/head = defender.get_organ(selected_zone)
+	if(!head)
+		if(!silence)
+			balloon_alert(attacker, "голова отсутствует!")
```
```suggestion
	if(!head && !silence)
		balloon_alert(attacker, "голова отсутствует!")
```

vovakons replied:
тут снизу return FALSE для !head, а показ балуна только если silence == TRUE

littleboobs commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -187,15 +187,63 @@
+	attacker.stop_pulling()
+	var/sound = pick('sound/weapons/knife_holster/throat_slice.ogg', 'sound/weapons/knife_holster/throat_slice2.ogg')
+	playsound(defender.loc, sound, 25, TRUE)
+	defender.apply_damage(2 * force, BRUTE, BODY_ZONE_HEAD, 0, TRUE, src, FALSE, FALSE, FALSE, TRUE, TRUE)
```
зачем?
```suggestion
	defender.apply_damage(2 * force, def_zone = BODY_ZONE_HEAD, sharp = TRUE, used_weapon = src)
```

littleboobs commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -187,15 +187,63 @@
+	if(!can_neck_cut(defender, attacker, FALSE))
+		return FALSE
+
+	attacker.visible_message(span_danger("[attacker] прикладывает нож к горлу [defender]!"), span_danger("Вы прикладываете нож к горлу [defender]!."))
```
балун ту вьюерс

littleboobs commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -187,15 +187,63 @@
+	var/sound = pick('sound/weapons/knife_holster/throat_slice.ogg', 'sound/weapons/knife_holster/throat_slice2.ogg')
+	playsound(defender.loc, sound, 25, TRUE)
+	defender.apply_damage(2 * force, BRUTE, BODY_ZONE_HEAD, 0, TRUE, src, FALSE, FALSE, FALSE, TRUE, TRUE)
+	attacker.visible_message(span_danger("[attacker] перерезает глотку [defender]!"), span_danger("Вы перерезаете глотку [defender]!"))
```
балун ту вьюерс

littleboobs commented on code/modules/mob/living/carbon/human/human_defense.dm:
```diff
@@ -674,6 +674,13 @@ emp_act
 		var/mob/living/carbon/human/H = user
 		dna.species.spec_attack_hand(H, src)
 
+/mob/living/carbon/human/click_alt(mob/user)
+	if(user == src)
+		dna.species.try_self_supress_bleeding(user)
+		return CLICK_ACTION_SUCCESS
+	return NONE
+
```
```suggestion
/mob/living/carbon/human/click_alt(mob/user)
	if(user != src)
		return NONE

	dna.species.try_self_supress_bleeding(user)
	return CLICK_ACTION_SUCCESS
```

littleboobs commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -520,6 +520,29 @@
+	user.balloon_alert(user, "кровотечение зажато!")
+	user.visible_message(
+			span_notice("[user] зажима[PLUR_ET_UT(user)] кровотечение на [target_limb.declent_ru(PREPOSITIONAL)]."),
+			ignored_mobs = user)
```
балун ту вьюерс

littleboobs commented on code/modules/power/cable_coil.dm:
```diff
@@ -139,6 +141,12 @@
 				cablecuff.icon_state = "cuff_[text_color]"
+		if("самодельный жгут (20)")
+			if(get_amount() < 20)
+				balloon_alert(user, "недостаточно [declent_ru(GENITIVE)]!")
```
можешь число в балун подставлять

littleboobs reviewed (changes requested):
Тяжело читать код не разбитый пустыми строками( Накинь их и балунов ту вьюерс, я пометил.

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
review fixes

vovakons committed:
rebuild tgui

vovakons committed:
fix linters issue

vovakons committed:
heal balance

Удалено использование травмкитов на ходу.
Использование медстаков по одному за раз (нельзя накликать больше).
Травкиты теперь лечат по 20, но больше количество использования.

vovakons committed:
use medical stack balloon alert text

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -17,6 +17,7 @@
 	var/bleedsuppress = 0
 	var/healverb = "bandage"
```
Удали за ненадобностью.

PlayerUnknown14 replied:
Удали переменную если не нужна.

vovakons committed:
Merge branch 'master220' of https://github.com/ss220-space/Paradise into bleeding-part-2

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Update tgui.bundle.js

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
rebuild tgui

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7691

dageavtobusnick committed:
Update tgui.bundle.js

dageavtobusnick committed:
Merge branch 'bleeding-part-2' of https://github.com/vovakons/Paradise into pr/7691

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
Merge branches 'bleeding-part-2' and 'bleeding-part-2' of github.com:vovakons/Paradise into bleeding-part-2

vovakons committed:
rebuild tgui

vovakons committed:
Merge branch 'master220' into bleeding-part-2

vovakons committed:
run tgui

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -299,15 +302,16 @@
 
 /obj/item/stack/medical/bruise_pack/advanced/update_icon_state()
-	icon_state = "traumakit_[amount]"
+	icon_state = "traumakit_[round_down((amount+1) / 2, 1)]"
```
```suggestion
	icon_state = "traumakit_[round_down((amount + 1) / 2, 1)]"
```

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -383,14 +390,14 @@
 
 /obj/item/stack/medical/ointment/advanced/update_icon_state()
-	icon_state = "burnkit_[amount]"
+	icon_state = "burnkit_[round_down((amount+1) / 2, 1)]"
```
```suggestion
	icon_state = "burnkit_[round_down((amount + 1) / 2, 1)]"
```

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+		deltimer(necrotize_warning_timer_id)
+		necrotize_warning_timer_id = null
+	if(!necrotize_timer_id)
+		return
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+
+/obj/item/tourniquet/attack(mob/living/target, mob/living/user, params, def_zone, skip_attack_anim = FALSE)
+	. = ATTACK_CHAIN_PROCEED
+	if(!ishuman(target))
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+	var/mob/living/carbon/human/human_target = target
+	var/obj/item/organ/external/affecting = human_target.get_organ(user.zone_selected)
+	var/obj/item/organ/external/addition_affecting = human_target.get_affecting_limb_bodypart(affecting)
+	if(affecting.tourniquet)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+		return .
+
+	var/selected_zone = user.zone_selected
+	if(human_target == user)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+			balloon_alert(user, "уже наложен жгут!")
+			return .
+		if(affecting_rechecked.is_robotic())
+			to_chat(human_target, span_danger("[capitalize(declent_ru(NOMINATIVE))] нельзя применить на протезе!"))
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+			return .
+		if(affecting_rechecked.is_robotic())
+			balloon_alert(user, "нельзя применить на протезе!")
+			return .
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+	target.UpdateDamageIcon()
+	update_icon()
+	necrotize_warning_timer_id = addtimer(CALLBACK(src, PROC_REF(necrotize_limbs_warning), target), necrotize_warning_duration, TIMER_STOPPABLE)
+	necrotize_timer_id = addtimer(CALLBACK(src, PROC_REF(necrotize_limbs), target), necrotize_duration, TIMER_STOPPABLE)
```
Огромный прок. Вынеси часть функционала в другие проки.

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+
+/obj/item/tourniquet/proc/necrotize_limbs_warning(mob/living/user)
+	if(!applyed_bodypart)
+		return
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+	if(applyed_bodypart)
+		applyed_bodypart.necrotize()
+	if(!applyed_addition_bodypart)
+		return
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -1730,12 +1760,13 @@ BODY SCANNERS
-		if(LAZYLEN(e.embedded_objects) || e.hidden)
+		if(bodypart.bleeding_amount > 0)
+			bled = "[round(bodypart.bleeding_amount, 0.01)] "
+		if(LAZYLEN(bodypart.embedded_objects) || bodypart.hidden)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -887,26 +905,38 @@ BODY SCANNERS
+				bleed_stat += "обильное "
+			if(bodypart.bleeding_amount <= bodypart.bleedsuppress)
+				bleed_stat += "остановленное "
+			scan_data += span_warning("Обнаружено [bleed_stat]кровотечение в [bodypart.declent_ru(PREPOSITIONAL)].")
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -739,30 +743,44 @@ BODY SCANNERS
+			bleeding += "  в [limb]"
+			if(bodypart.bleeding_amount <= bodypart.bleedsuppress)
+				bleeding += " – остановлено"
+			bleedingList += bleeding
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/machinery/adv_med.dm:
```diff
@@ -518,10 +521,22 @@
+					bleeding += "артериальное "
+				else if(e.has_heavy_bleeding())
+					bleeding += "обильное "
+				bleeding += "кровотечение"
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/machinery/adv_med.dm:
```diff
@@ -518,10 +521,22 @@
-				internal_bleeding = "<br>Внутреннее кровотечение"
+				if(bleeding == "")
+					internal_bleeding = "<br>"
+				internal_bleeding += "Внутреннее кровотечение"
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/datums/keybindings/mob.dm:
```diff
@@ -51,6 +51,16 @@
+				else
+					grabber.right_hand_bleed_suppress_lib = null
+				grabber.update_hands_HUD()
+				return TRUE
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/_onclick/other_mobs.dm:
```diff
@@ -75,6 +75,9 @@
 		if(!limb.is_usable())
 			to_chat(src, span_warning("Ваша [hand ? "левая рука" : "правая рука"] слишком травмирована."))
 			return FALSE
+		if((hand == ACTIVE_HAND_RIGHT && right_hand_bleed_suppress_lib) || (hand == ACTIVE_HAND_LEFT && left_hand_bleed_suppress_lib))
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/_onclick/hud/screen_objects.dm:
```diff
@@ -645,6 +647,11 @@
+	else if(!hand_blocked && isliving(user))
+		var/mob/living/grabber = user
+		var/suppress_target_bodypart = left_hand ? grabber.left_hand_bleed_suppress_lib : grabber.right_hand_bleed_suppress_lib
+		if(suppress_target_bodypart)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+		tourniquet.applyed_bodypart = null
+		if(tourniquet.applyed_addition_bodypart)
+			tourniquet.applyed_addition_bodypart.tourniquet = null
+			tourniquet.applyed_addition_bodypart = null
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+		return TRUE
+	if(zone_selected == BODY_ZONE_HEAD)
+		return TRUE
+	return FALSE
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+	for(var/obj/item/organ/external/bodypart as anything in bodyparts)
+		if(bodypart.tourniquet)
+			return TRUE
+	return FALSE
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,239 @@
+	applyed_bodypart = null
+	if(applyed_addition_bodypart)
+		applyed_addition_bodypart.tourniquet = null
+		applyed_addition_bodypart = null
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,62 @@
+			. = ATTACK_CHAIN_BLOCKED_ALL
+			neck_cut(target, user)
+			force = default_force
+			return
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,62 @@
-			if(MA.neck_cut(target, user))
-				return ATTACK_CHAIN_PROCEED_SUCCESS
+	if(user.zone_selected == BODY_ZONE_HEAD && user.a_intent == INTENT_HARM)
+		var/datum/martial_art/throwing/MA = user?.mind?.martial_art
```
имя переменной

dageavtobusnick commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,62 @@
+	if(!head)
+		if(!silence)
+			balloon_alert(attacker, "голова отсутствует!")
+		return FALSE
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,62 @@
+	if(defender.blood_volume > BLOOD_VOLUME_SURVIVE)
+		defender.blood_volume = max(0, defender.blood_volume - 0.25 * (BLOOD_VOLUME_NORMAL - BLOOD_VOLUME_SURVIVE)) //-25% of max blood volume
+		for(var/i in 1 to 2)
+			var/obj/effect/decal/cleanable/blood/B = new(defender.loc)
```
имя переменной

dageavtobusnick commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,62 @@
+			balloon_alert(attacker, "не получилось!")
+			return FALSE
+	if(defender.blood_volume > BLOOD_VOLUME_SURVIVE)
+		defender.blood_volume = max(0, defender.blood_volume - 0.25 * (BLOOD_VOLUME_NORMAL - BLOOD_VOLUME_SURVIVE)) //-25% of max blood volume
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,62 @@
+		for(var/i in 1 to 2)
+			var/obj/effect/decal/cleanable/blood/B = new(defender.loc)
+			B.blood_DNA[defender.dna.unique_enzymes] = defender.dna.blood_type
+			step(B, pick(GLOB.alldirs))
```
отделяем for от кода и других ифов пустой строкой(исключение - начало прока).

dageavtobusnick commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -340,6 +341,9 @@
 
 		for(var/obj/item/embedded as anything in bodypart.embedded_objects)
 			status_list += "\t <a href='byond://?src=[UID()];embedded_object=[embedded.UID()];embedded_limb=[bodypart.UID()]' class='warning'>В ваш[GEND_EM_EI_EM_IH(bodypart)] [bodypart.declent_ru(GENITIVE)] застрял[GEND_A_O_I(embedded)] [icon2html(embedded, src)] [embedded.declent_ru(NOMINATIVE)]!</a>"
+		if(bodypart.tourniquet && bodypart == bodypart.tourniquet.applyed_bodypart)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -366,16 +366,24 @@
+			msg += span_warning("[capitalize(GEND_HIS_HER(src))] [bodypart.declent_ru(NOMINATIVE)] перевязан[GEND_A_O_Y(src)] чем-то окровавленным.\n")
+		else if(bodypart.has_arterial_bleeding())
+			msg += span_warning(span_bold("Из [GEND_HIS_HER(src)] [bodypart.declent_ru(GENITIVE)] хлещет кровь!\n"))
+		else if(bodypart.has_heavy_bleeding())
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/mob/living/carbon/human/human_damage.dm:
```diff
@@ -479,6 +479,26 @@ This function restores all organs.
+		if(BODY_ZONE_PRECISE_L_FOOT)
+			return get_organ(BODY_ZONE_L_LEG)
+		if(BODY_ZONE_PRECISE_R_FOOT)
+			return get_organ(BODY_ZONE_R_LEG)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/mob/living/carbon/human/human_organs.dm:
```diff
@@ -146,6 +146,13 @@
+	for(var/obj/item/organ/external/bodypart as anything in bodyparts)
+		if(bodypart.has_arterial_bleeding())
+			bleeding_bodyparts += bodypart
+	return bleeding_bodyparts
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока). И for тоже.

dageavtobusnick commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -508,6 +508,30 @@
+	var/obj/item/organ/external/target_limb = user.get_organ(user.zone_selected)
+	if(target_limb.bleeding_amount <= 0)
+		user.balloon_alert(user, "кровотечения нет!")
+		return
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -508,6 +508,30 @@
+	if(user.hand == ACTIVE_HAND_LEFT)
+		user.left_hand_bleed_suppress_lib = target_limb
+	else
+		user.right_hand_bleed_suppress_lib = target_limb
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/power/cable_coil.dm:
```diff
@@ -132,6 +134,12 @@
+		if("самодельный жгут (20)")
+			if(get_amount() < 20)
+				balloon_alert(user, "недостаточно [20 - get_amount()] [declent_ru(GENITIVE)]!")
+			if(use(20))
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/power/cable_coil.dm:
```diff
@@ -132,6 +134,12 @@
 						text_color = "white"
 
 				cablecuff.icon_state = "cuff_[text_color]"
+		if("самодельный жгут (20)")
```
Выборы дефайнизировать

dageavtobusnick commented on code/modules/surgery/abstract_steps.dm:
```diff
@@ -274,6 +274,10 @@
 		return TRUE
+	if(affected.has_arterial_bleeding())
+		return TRUE
+	if(affected.bleeding_amount > 0)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/blood.dm:
```diff
@@ -123,12 +186,34 @@
-		bleed(bleed_rate * additional_bleed_mod)
+		bleed(bleed_rate * additional_bleed_mod * speed_by_volume * speed_by_bodytemperature)
+	// make bloodsplatter for arterial bleeding
+	if(has_arterial_bleed)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/blood.dm:
```diff
@@ -123,12 +186,34 @@
+		return BLOODLOSS_SPEED_BY_TEMP_MAX
+	if(bodytemperature <= T0C)
+		return BLOODLOSS_SPEED_BY_TEMP_MIN
+	var/temperature_percent = clamp((bodytemperature - T0C) / (BODYTEMP_NORMAL * 0.75 - T0C), 0, 1)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -163,6 +174,9 @@
 	QDEL_LIST(embedded_objects)
 	QDEL_NULL(hidden)
+	if(tourniquet && !QDELETED(tourniquet))
+		QDEL_NULL(tourniquet)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -426,6 +442,10 @@
+	if(owner && HAS_TRAIT(owner, TRAIT_NO_BLOOD))
+		return
+	if(has_arterial_bleeding())
+		return //has arterial bleeding, no more bleedings
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -665,6 +685,18 @@ Note that amputating the affected organ does in fact remove the infection from t
+	if(arterial_bleeding(silent))
+		add_attack_logs(owner, null, "Suffered arterial bleeding to [src](Damage: [inflicted_damage], Organ HP: [max_damage - (brute_dam + burn_dam) ])")
+		return TRUE
+	return FALSE
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -951,6 +983,46 @@ Note that amputating the affected organ does in fact remove the infection from t
+			return FALSE
+	if(is_robotic())
+		return FALSE
+	if(has_arterial_bleeding() || cannot_arterial_bleed)
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -951,6 +983,46 @@ Note that amputating the affected organ does in fact remove the infection from t
+	if(owner)
+		if(HAS_TRAIT(owner, TRAIT_GODMODE))
+			return FALSE
+		if(HAS_TRAIT(owner, TRAIT_NO_BLOOD))
```
Может это в 1 проверку?

dageavtobusnick commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -951,6 +983,46 @@ Note that amputating the affected organ does in fact remove the infection from t
+	if(!has_arterial_bleeding())
+		return FALSE
+	bleeding_amount = 0.1 * max_bleeding_amount //low bleeding exists after stop arterial bleed
+	return TRUE
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

dageavtobusnick commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -1248,9 +1320,17 @@ Note that amputating the affected organ does in fact remove the infection from t
 
+/obj/item/organ/external/proc/remove_tourniquet(atom/drop_loc)
+	if(!tourniquet)
+		return
```
отделяем ифы от кода и других ифов пустой строкой(исключение - начало прока)

vovakons committed:
review fixes

vovakons committed:
linter fix

PlayerUnknown14 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -406,6 +406,10 @@ BODY SCANNERS
 
+	if(scan_data["bleedingList"])
+		for(var/bleeding in scan_data["bleedingList"])
+			P.header += span_red("Кровотечение в [bleeding].<br>")
```
На русском выводит, надеюсь?

PlayerUnknown14 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -1731,11 +1765,14 @@ BODY SCANNERS
 			AN = "None:"
-		dat += "<td>[e.name]</td><td>[e.burn_dam]</td><td>[e.brute_dam]</td><td>[robot][bled][AN][splint][open][infected][imp][internal_bleeding][lung_ruptured]</td>"
+
+		dat += "<td>[bodypart.name]</td><td>[bodypart.burn_dam]</td><td>[bodypart.brute_dam]</td><td>[robot][bled][AN][splint][open][infected][imp][internal_bleeding][lung_ruptured]</td>"
```
деклент бодипарту

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -76,6 +74,7 @@
 				span_notice("[user] применя[PLUR_ET_YUT(user)] [declent_ru(NOMINATIVE)] на [human_target]."),
 				span_notice("Вы начинаете применять [declent_ru(NOMINATIVE)] на [human_target]..."),
 			)
+			user.balloon_alert_to_viewers("применя[PLUR_ET_YUT(user)] [declent_ru(ACCUSATIVE)]...", "применение [declent_ru(GENITIVE)]...");
```
human_target.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -52,11 +53,8 @@
-				span_notice("Вы начинаете применять [declent_ru(NOMINATIVE)] на себе..."),
-			)
-			if(!do_after(human_target, self_delay, human_target, DA_IGNORE_USER_LOC_CHANGE | DA_IGNORE_LYING))
+			user.balloon_alert_to_viewers("применя[PLUR_ET_YUT(user)] [declent_ru(ACCUSATIVE)]...", "применение [declent_ru(GENITIVE)]...");
```
human_target.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+// MARK: Tourniquet
+/obj/item/tourniquet
+	name = "tourniquet"
+	desc = "Обычный жгут для остановки всех видов кровотечений на конечностях. При долгом ношении конечность может отмереть. Можно наложить только на конечности."
```
```suggestion
	desc = "Медицинский жгут для экстренной остановки артериального и венозного кровотечения на конечностях. Не предназначен для наложения на другие части тела. Длительное использование без последующей медицинской помощи ведёт к некрозу тканей."
```

PlayerUnknown14 replied:
```suggestion
	desc = "Медицинский жгут для экстренной остановки артериального и венозного кровотечения на конечностях. Длительное использование без последующей медицинской помощи ведёт к некрозу тканей. Несколько устарел морально и технически, но со своей задачей справится."
```

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+		return .
+
+	if(!acceptable_zone(user.zone_selected))
+		balloon_alert(user, "недопустимая зона!")
```
```suggestion
		balloon_alert(user, "не является конечностью!")
```

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+	var/obj/item/organ/external/affecting = human_target.get_organ(user.zone_selected)
+	var/obj/item/organ/external/addition_affecting = human_target.get_affecting_limb_bodypart(affecting)
+	if(affecting.tourniquet)
+		balloon_alert(user, "уже наложен жгут!")
```
уже наложено!

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+
+/obj/item/tourniquet/proc/apply_to_self(mob/living/carbon/human/user, obj/item/organ/external/affecting, obj/item/organ/external/addition_affecting)
+	var/selected_zone = user.zone_selected
+	balloon_alert(user, "применение [declent_ru(GENITIVE)]...")
```
Балун_ту_вьюерс

Для юзера - "наложение [...]..."

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+		return
+
+	if(affecting_rechecked.is_robotic())
+		to_chat(user, span_danger("[capitalize(declent_ru(NOMINATIVE))] нельзя применить на протезе!"))
```
балун "неорганическая конечность!"

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+
+	var/obj/item/organ/external/affecting_rechecked = user.get_organ(selected_zone)
+	if(!affecting_rechecked)
+		balloon_alert(user, "часть тела отсутствует!")
```
конечность

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+		return
+
+	if(affecting_rechecked.tourniquet)
+		balloon_alert(user, "уже наложен жгут!")
```
.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+
+/obj/item/tourniquet/proc/apply_to_other(mob/living/user, mob/living/carbon/human/human_target, obj/item/organ/external/affecting, obj/item/organ/external/addition_affecting)
+	var/selected_zone = user.zone_selected
+	user.balloon_alert_to_viewers("применение [declent_ru(GENITIVE)]...")
```
.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+/obj/item/tourniquet/proc/apply_to_other(mob/living/user, mob/living/carbon/human/human_target, obj/item/organ/external/affecting, obj/item/organ/external/addition_affecting)
+	var/selected_zone = user.zone_selected
+	user.balloon_alert_to_viewers("применение [declent_ru(GENITIVE)]...")
+	user.visible_message(
```
.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+
+	var/obj/item/organ/external/affecting_rechecked = human_target.get_organ(selected_zone)
+	if(!affecting_rechecked)
+		balloon_alert(user, "часть тела отсутствует!")
```
.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+		return
+
+	if(affecting_rechecked.tourniquet)
+		balloon_alert(user, "уже наложен жгут!")
```
.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+		return
+
+	if(affecting_rechecked.is_robotic())
+		balloon_alert(user, "нельзя применить на протезе!")
```
.

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+	if(!applyed_bodypart)
+		return
+
+	balloon_alert(user, "[applyed_bodypart.declent_ru(NOMINATIVE)] онемела!")
```
ваш[гендерайз] [...] немеет!

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+		return
+
+	balloon_alert(user, "[applyed_bodypart.declent_ru(NOMINATIVE)] онемела!")
+	to_chat(user, span_danger("Вы ощущаете, как ваша [applyed_bodypart.declent_ru(NOMINATIVE)] теряет чувствительность!"))
```
убери

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+
+/obj/item/tourniquet/makeshift
+	name = "makeshift tourniquet"
+	desc = "Самодельный жгут для остановки всех видов кровотечений на конечностях. При долгом ношении конечность может отмереть. Можно наложить только на конечности."
```
```suggestion
	desc = "Импровизированный жгут для временной остановки кровотечения на конечностях. Жутко неудобный, но со своей задачей справится. Не предназначен для длительного использования."
```

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -508,6 +508,35 @@
+	user.visible_message(
+			span_notice("[user] зажима[PLUR_ET_UT(user)] кровотечение на [target_limb.declent_ru(PREPOSITIONAL)]."),
+			ignored_mobs = user)
+	user.balloon_alert_to_viewers("зажима[PLUR_ET_YUT(user)] кровотечение на [target_limb.declent_ru(PREPOSITIONAL)]", "зажатие кровотечения на [target_limb.declent_ru(PREPOSITIONAL)]");
```
зажимает **рану**

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -508,6 +508,35 @@
+	user.balloon_alert(user, "кровотечение зажато!")
+	user.visible_message(
+			span_notice("[user] зажима[PLUR_ET_UT(user)] кровотечение на [target_limb.declent_ru(PREPOSITIONAL)]."),
+			ignored_mobs = user)
```
убрать

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +656,271 @@
+
+/obj/item/tourniquet/advanced
+	name = "advanced tourniquet"
+	desc = "Турникет для остановки всех видов кровотечений на конечностях. При долгом ношении конечность может отмереть. Можно наложить только на конечности."
```
```suggestion
	desc = "Медицинский турникет нового поколения для экстренной остановки артериального и венозного кровотечения на конечностях. Оснащён механизмом контроля давления, что повышает удобство использования и его эффективность по сравнению с ранними аналогами. Длительное использование без последующей медицинской помощи ведёт к некрозу тканей."
```

PlayerUnknown14 commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,69 @@
+	if(!can_neck_cut(defender, attacker, FALSE))
+		return FALSE
+
+	attacker.visible_message(span_danger("[attacker] прикладывает нож к горлу [defender]!"), span_danger("Вы прикладываете нож к горлу [defender]!."))
```
балун_ту_вьюверс, attacker и defender убрать в нём

Визибл оставь.

PlayerUnknown14 commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,69 @@
+
+	var/obj/item/organ/external/head = defender.get_organ(attacker.zone_selected)
+	if(!head.has_arterial_bleeding())
+		if(!head.arterial_bleeding())
```
смысл такой проверки?

PlayerUnknown14 replied:
смысл такой проверки?

PlayerUnknown14 commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,69 @@
+	var/obj/item/organ/external/head = defender.get_organ(attacker.zone_selected)
+	if(!head.has_arterial_bleeding())
+		if(!head.arterial_bleeding())
+			balloon_alert(attacker, "не получилось!")
```
defender.

PlayerUnknown14 commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,69 @@
+	var/sound = pick('sound/weapons/knife_holster/throat_slice.ogg', 'sound/weapons/knife_holster/throat_slice2.ogg')
+	playsound(defender.loc, sound, 25, TRUE)
+	defender.apply_damage(2 * force, def_zone = BODY_ZONE_HEAD, sharp = TRUE, used_weapon = src)
+	attacker.balloon_alert_to_viewers("перерезает глотку [defender]", "горло перерезано!");
```
плюралайз, убрать дефендер

PlayerUnknown14 replied:
Ниже пропиши визибл.

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -366,16 +366,25 @@
+			continue
+		if(bodypart.tourniquet.applyed_bodypart != bodypart)
+			continue
+		msg += span_warning("<a href='byond://?src=[UID()];tourniquet_object=[bodypart.tourniquet.UID()];limb=[bodypart.UID()]' class='warning'>[GEND_HIS_HER_CAP(src)] [bodypart.declent_ru(NOMINATIVE)] перевязан[GEND_A_O_Y(src)] [icon2html(bodypart.tourniquet, src)] [bodypart.tourniquet.declent_ru(INSTRUMENTAL)]!</a>\n")
```
```suggestion
		msg += span_warning("<a href='byond://?src=[UID()];tourniquet_object=[bodypart.tourniquet.UID()];limb=[bodypart.UID()]' class='warning'>[GEND_HIS_HER_CAP(src)] [bodypart.declent_ru(NOMINATIVE)] пережат[GEND_A_O_Y(src)] [icon2html(bodypart.tourniquet, src)] [bodypart.tourniquet.declent_ru(INSTRUMENTAL)]!</a>\n")
```

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -366,16 +366,25 @@
+		else if(bodypart.has_arterial_bleeding())
+			msg += span_warning(span_bold("Из [GEND_HIS_HER(src)] [bodypart.declent_ru(GENITIVE)] хлещет кровь!\n"))
+		else if(bodypart.has_heavy_bleeding())
 			msg += span_warning(span_bold("У н[GEND_HIS_HER(src)] обильно кровоточ[PLUR_IT_AT(src)] [bodypart.declent_ru(NOMINATIVE)]!\n"))
```
```suggestion
			msg += span_warning(span_bold("[GEND_CAP_HIS_HER(src)] [bodypart.declent_ru(NOMINATIVE)] обильно кровоточ[PLUR_IT_AT(src)]!\n"))
```
Синтаксис первого гендерайза не помню, поправишь.

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -366,16 +366,25 @@
+		else if(bodypart.has_heavy_bleeding())
 			msg += span_warning(span_bold("У н[GEND_HIS_HER(src)] обильно кровоточ[PLUR_IT_AT(src)] [bodypart.declent_ru(NOMINATIVE)]!\n"))
 		else
 			msg += span_warning(span_bold("У н[GEND_HIS_HER(src)] кровоточ[PLUR_IT_AT(src)] [bodypart.declent_ru(NOMINATIVE)]!\n"))
```
Сейм переписать

PlayerUnknown14 commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -341,6 +342,9 @@
 			status_list += "\t <a href='byond://?src=[UID()];embedded_object=[embedded.UID()];embedded_limb=[bodypart.UID()]' class='warning'>В ваш[GEND_EM_EI_EM_IH(bodypart)] [bodypart.declent_ru(GENITIVE)] застрял[GEND_A_O_I(embedded)] [icon2html(embedded, src)] [embedded.declent_ru(NOMINATIVE)]!</a>"
 
+		if(bodypart.tourniquet && bodypart == bodypart.tourniquet.applyed_bodypart)
+			status_list += "\t <a href='byond://?src=[UID()];tourniquet_object=[bodypart.tourniquet.UID()];limb=[bodypart.UID()]' class='warning'>Ваша [bodypart.declent_ru(NOMINATIVE)] перевязан[GEND_A_O_I(bodypart)] [icon2html(bodypart.tourniquet, src)] [bodypart.tourniquet.declent_ru(INSTRUMENTAL)]!</a>"
```
Гендерайз в начале

Перевязано - пережато

PlayerUnknown14 commented on code/modules/power/cable_coil.dm:
```diff
@@ -132,17 +141,29 @@
+
+		if(CABLE_CRAFT_TOURNIQUET)
+			if(get_amount() < 20)
+				balloon_alert(user, "недостаточно [20 - get_amount()] [declent_ru(GENITIVE)]!")
```
```suggestion
				balloon_alert(user, "недостаточно проводов!")
```

PlayerUnknown14 commented on code/modules/surgery/organs/organ_external.dm:
```diff
@@ -951,6 +994,48 @@ Note that amputating the affected organ does in fact remove the infection from t
+
+	if(owner && !silent)
+		owner.custom_pain("Из ваш[GEND_HIS_HER(src)] [declent_ru(GENITIVE)] хлещет кровь!")
+		owner.visible_message(span_warning("Из [declent_ru(GENITIVE)] [owner] хлещет кровь!"))
```
деклент овнеру

vovakons committed:
localization review fixes

vovakons committed:
return tourniquets

vovakons committed:
autodrop hand bleedsuppress

vovakons committed:
remove ballon alert

vovakons committed:
linter fixes

vovakons committed:
linter fixes 2

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+
+	user.drop_item_ground(src)
+	src.forceMove(affecting)
+	user.balloon_alert_to_viewers("турникет наложен")
```
```suggestion
	balloon_alert(user, "турникет наложен")
```

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+	user.visible_message(
+		span_notice("[user] начина[PLUR_ET_UT(user)] применять на себе [declent_ru(ACCUSATIVE)]."),
+		ignored_mobs = user
+	)
```
убрать

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+
+/obj/item/tourniquet/proc/apply_to_self(mob/living/carbon/human/user, obj/item/organ/external/affecting, obj/item/organ/external/addition_affecting)
+	var/selected_zone = user.zone_selected
+	user.balloon_alert_to_viewers("применение [declent_ru(GENITIVE)]...", "наложение [declent_ru(GENITIVE)]...")
```
применя[ет] [...] на себя...

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+		return
+
+	if(affecting_rechecked.tourniquet)
+		balloon_alert(user, "уже наложен турникет")
```
```suggestion
		balloon_alert(user, "турникет уже наложен!")
```

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+
+/obj/item/tourniquet/proc/apply_to_other(mob/living/user, mob/living/carbon/human/human_target, obj/item/organ/external/affecting, obj/item/organ/external/addition_affecting)
+	var/selected_zone = user.zone_selected
+	user.balloon_alert_to_viewers("применение [declent_ru(GENITIVE)]")
```
убрать

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+/obj/item/tourniquet/proc/apply_to_other(mob/living/user, mob/living/carbon/human/human_target, obj/item/organ/external/affecting, obj/item/organ/external/addition_affecting)
+	var/selected_zone = user.zone_selected
+	user.balloon_alert_to_viewers("применение [declent_ru(GENITIVE)]")
+	user.balloon_alert_to_viewers("применя[PLUR_ET_YUT(user)] [declent_ru(ACCUSATIVE)]...", "применение [declent_ru(GENITIVE)]...")
```
```suggestion
	human_target.balloon_alert_to_viewers("применя[PLUR_ET_YUT(user)] [declent_ru(ACCUSATIVE)] на цели...", "применение [declent_ru(GENITIVE)] на цели...")
```

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+		return
+
+	var/obj/item/organ/external/affecting_rechecked = human_target.get_organ(selected_zone)
+	if(!affecting_rechecked)
+		balloon_alert(user, "часть тела отсутствует")
+		return
+
+	if(affecting_rechecked.tourniquet)
+		balloon_alert(user, "уже наложен турникет")
+		return
+
+	if(affecting_rechecked.is_robotic())
+		balloon_alert(user, "нельзя применить на протезе")
+		return
```
сообщения возьми из прока выше

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+	if(!applyed_bodypart)
+		return FALSE
+
+	user.balloon_alert_to_viewers("снятие турникета...")
```
```suggestion
	balloon_alert(user, "снятие турникета...")
```

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+
+	stop_apply_timers()
+	user.put_in_any_hand_if_possible(src)
+	user.balloon_alert_to_viewers("турникет снят")
```
```suggestion
	balloon_alert(user, "турникет снят")
```

PlayerUnknown14 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -643,13 +654,263 @@
+	return FALSE
+
+/mob/living/carbon/human/proc/cut_all_tourniquets(mob/living/user)
+	src.balloon_alert_to_viewers("турникеты срезаны!")
```
Балун лучше вызывать после вызова этого прока, не здесь.

PlayerUnknown14 commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -176,15 +176,63 @@
+	if(!can_neck_cut(defender, attacker, FALSE))
+		return FALSE
+
+	attacker.balloon_alert_to_viewers("прикладывает нож к горлу!", "вы прикладываете нож к горлу!")
```
```suggestion
	defender.balloon_alert_to_viewers("прикладывает нож к горлу!", "вы прикладываете нож к горлу!")
```

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -508,6 +508,34 @@
+		user.balloon_alert(user, "кровотечения нет!")
+		return
+
+	user.balloon_alert_to_viewers("зажима[PLUR_ET_YUT(user)] рану на [target_limb.declent_ru(PREPOSITIONAL)]", "зажатие раны на [target_limb.declent_ru(PREPOSITIONAL)]");
```
Тут точно не будет выводиться что-то типа "зажимает рану на левой руке вульпканина"? Есть сомнения насчёт возвращаемого ру_нейма конечности.

vovakons replied:
норм

vovakons replied:
<img width="323" height="131" alt="image" src="https://github.com/user-attachments/assets/f96c9fcf-7213-4ecf-a7a0-1a536c9c9d0e" />

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -366,19 +366,28 @@
+			msg += span_warning(span_bold("[GEND_HIS_HER_CAP(src)] [bodypart.declent_ru(NOMINATIVE)] обильно кровоточ[PLUR_IT_AT(src)]!\n"))
 		else
-			msg += span_warning(span_bold("У н[GEND_HIS_HER(src)] кровоточ[PLUR_IT_AT(src)] [bodypart.declent_ru(NOMINATIVE)]!\n"))
+			msg += span_warning(span_bold("[GEND_HIS_HER_CAP(src)] [bodypart.declent_ru(NOMINATIVE)] кровоточ[PLUR_IT_AT(src)]!\n"))
```
```suggestion
			msg += span_warning(span_bold("[GEND_HIS_HER_CAP(src)] [bodypart.declent_ru(NOMINATIVE)] кровоточ[PLUR_IT_AT(bodypart)]!\n"))
```

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -366,19 +366,28 @@
+		else if(bodypart.has_arterial_bleeding())
+			msg += span_warning(span_bold("Из [GEND_HIS_HER(src)] [bodypart.declent_ru(GENITIVE)] хлещет кровь!\n"))
+		else if(bodypart.has_heavy_bleeding())
+			msg += span_warning(span_bold("[GEND_HIS_HER_CAP(src)] [bodypart.declent_ru(NOMINATIVE)] обильно кровоточ[PLUR_IT_AT(src)]!\n"))
```
```suggestion
			msg += span_warning(span_bold("[GEND_HIS_HER_CAP(src)] [bodypart.declent_ru(NOMINATIVE)] обильно кровоточ[PLUR_IT_AT(bodypart)]!\n"))
```

PlayerUnknown14 commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -366,19 +366,28 @@
-			msg += span_warning("У н[GEND_HIS_HER(src)] [bodypart.declent_ru(NOMINATIVE)] перевязан[GEND_A_O_Y(src)] чем-то окровавленным.\n")
-		else if(high_bleeding)
-			msg += span_warning(span_bold("У н[GEND_HIS_HER(src)] обильно кровоточ[PLUR_IT_AT(src)] [bodypart.declent_ru(NOMINATIVE)]!\n"))
+			msg += span_warning("[capitalize(GEND_HIS_HER(src))] [bodypart.declent_ru(NOMINATIVE)] перевязан[GEND_A_O_Y(src)] чем-то окровавленным.\n")
```
```suggestion
			msg += span_warning("[capitalize(GEND_HIS_HER(src))] [bodypart.declent_ru(NOMINATIVE)] перевязан[GEND_A_O_Y(bodypart)] чем-то окровавленным.\n")
```

PlayerUnknown14 commented on code/modules/vending/security.dm:
```diff
@@ -59,6 +60,7 @@
 		/obj/item/flashlight/sectaclight = 299,
 		/obj/item/grenade/smokebomb = 249,
 		/obj/item/stack/medical/bruise_pack/military = 299,
+		/obj/item/tourniquet/advanced = 399,
```
А надо? Я бы на твоём месте положил 1-2 в стартовую аптечку и всё на этом.

vovakons committed:
localization fixes

vovakons committed:
disable internal bleeding in hands and foots
